### DISCUSSION TITLE
PerformanceData Postprocessor Error

### DISCUSSION THREAD

**Discussion by singhgp4321** [idaholab/moose#16625] [Q&A General] [answered]
I am trying to obtain the jacobian compute metrics for a simulation in Bison app using the PerformanceData postprocessor and keep getting the following error:

*Unknown section_name: FEProblem::computeJacobianInternal in PerfGraph::getTime()
If you are attempting to retrieve the root use "Root".*

If I change the `data_type` to `calls` in the postprocessor block, it shows the same error in `PerfGraph::getNumCalls()`.

I tried to debug it. Looking at the top frame when it is about to error out: it tries to find `section_name` (holding the string `FEProblem::computeJacobianInternal`) inside the container `_section_time`. The `_section_time` container seems to have many objects except `FEProblem::computeJacobianInternal`.

When I remove this postprocessor and turn on the `perf_graph` I see the `computeJacobianInternal` related information in the generated performance log table - indicating the jacobian metrics are being determined successfully but postprocessor is not able to retrieve that information.

I am wondering why is the PerformanceData postprocessor failing to retrieve the jacobian compute information.

Thanks,
Gyan

**Comment by aeslaughter** [comment]
@friedmud Can you help with this?

**Comment by singhgp4321** [comment]
Just wondering if someone could help with this. Let me know if you need any information.

**Comment by loganharbour** [comment]
`PerformanceData` is deprecrated. You should use `PerfGraphData` instead.

I believe this is what you're looking for:
```
[Postprocessors]
  [jacobian_time]
    type = PerfGraphData
    section_name = 'FEProblem::computeJacobianInternal'
    data_type = TOTAL
  []
[]
```

See here for more information: https://mooseframework.inl.gov/source/postprocessors/PerfGraphData.html

**Reply by singhgp4321** [reply]
Actually I am using the `PerfGraphData` and not the `PerformanceData` object. I got confused in naming due to this page (https://mooseframework.inl.gov/source/postprocessors/PerformanceData.html).

**Reply by loganharbour** [reply]
This is interesting. I added the above to `test/tests/kernels/simple_diffusion/simple_diffusion.i` and got the intended results.

Can you please open an issue within the BISON repository (do not do it here) with the input that you are running? Respond here when you do and I'll take a look there.

**Comment by loganharbour** [comment, marked as answer]
Alright, found the issue. Postprocessors by default run on `TIMESTEP_END`. If the section that you are executing has not been called yet before `TIMESTEP_END`, it doesn't exist it'll error out that it can't find the section.

For the time being, your solution is to set `execute_on` to something on your own that occurs _after_ the section has been called, which in your case is almost anything but `TIMESTEP_END`. Something like `NONLINEAR` or `FINAL`.

This isn't a very friendly error. We actually register the section very early on (well before `INITIAL`), therefore we do know that it exists. I'll work on a change that makes this more meaningful. I'd prefer that it just return 0 in all cases if it has been registered but has not been called yet.

**Comment by loganharbour** [comment]
#16704 will resolve this issue in the future - valid sections will return zero if they have not ran yet.

**Reply by singhgp4321** [reply]
Thanks Logan! Appreciate it very much!
